web/src/engine/websites/ColoredManga.ts

```diff
@@ -0,0 +1,120 @@
+    };
+}
+
+@Common.MangasSinglePageCSS('/manga', 'div#themes_outside__WCut6 a:not([id])', MangaInfoExtractor)
+
+export default class extends DecoratableMangaScraper {
+    private readonly apiUrl = `${this.URI.origin}/api/`;
+    private readonly mangaRegexp = new RegExp(`^${this.URI.origin}/manga/([^/]+)$`);
+
+    public constructor() {
+        super('coloredmanga', 'Colored Manga', 'https://coloredmanga.net', Tags.Media.Manhwa, Tags.Media.Manga, Tags.Language.English, Tags.Source.Scanlator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return this.mangaRegexp.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const mangaSlug = url.match(this.mangaRegexp)[1];
+        const { name } = await this.GetMangaData(mangaSlug);
+        return new Manga(this, provider, mangaSlug, name);
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const mangaData = await this.GetMangaData(manga.Identifier);
+        return this.ExtractChaptersData(mangaData).map(chapter => {
+            const title = [
+                chapter.number,
+                chapter.title ? '-' : '',
+                chapter.title,
+            ].join(' ').trim();
+            return new Chapter(this, manga, chapter.id, title);
+        });
+    }
+
+    private ExtractChaptersData(mangaData: APIManga): APIChapter[] {
+        return mangaData.chapters.length > 0 ? mangaData.chapters : mangaData.volume.reduce((chaptersAccumulator: APIChapter[], currentVolume) => {
+            currentVolume.chapters.forEach(chapter => chapter.volume = currentVolume.number);
+            chaptersAccumulator.push(...currentVolume.chapters);
+            return chaptersAccumulator;
+        }, []);
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const mangaData = await this.GetMangaData(chapter.Parent.Identifier);
+        const { totalImage, volume } = this.ExtractChaptersData(mangaData).find(item => item.id === chapter.Identifier);
+        const path = volume ? `/images/content/${mangaData.name}/${volume}/${chapter.Title}` : `/images/content/${mangaData.name}/${chapter.Title}`;
+        return new Array(totalImage).fill(0).map((_, index) => new Page(this, chapter, this.URI, { path, number: (index + 1).toString().padStart(4, '0') }));
+    }
+
+    public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
+
+        return await this.imageTaskPool.Add(async () => {
+
+            const request = this.CreateRequest('dynamicImages', {
+                path: page.Parameters['path'] as string,
+                number: page.Parameters['number'] as string
+            });
+
+            const { image } = await FetchJSON<APIImage>(request);
+            const response = await fetch(image);
```

Comment:
Will this work for `image` > 2MB?
There were some issues in HakuNeko about this:
- https://github.com/manga-download/hakuneko/pull/4336#discussion_r791682275
- https://github.com/manga-download/hakuneko/pull/4783#discussion_r912200805

Reply:
Is the limitation inherent to the fetch api or for data uri? 
In the first case we can convert it by ourselves , in the second case its the website problem 🤷

Reply:
[Length limitations](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs#length_limitations)

> Browsers are not required to support any particular maximum length of data. For example, the Opera 11 browser limited URLs to 65535 characters long which limits data URLs to 65529 characters (65529 characters being the length of the encoded data, not the source, if you use the plain data:, without specifying a MIME type). Firefox version 97 and newer supports data URLs of up to **32MB** (before 97 the limit was close to 256MB). Chromium objects to URLs over **512MB**, and Webkit (Safari) to URLs over **2048MB**.

Shouldnt it be fine?

Reply:
Tested with a 5mb dataURI 👍 

![image](https://github.com/user-attachments/assets/02ca4f40-1bc5-443a-b528-08737dc6aca9)

Reply:
Was this tested with both, the `NW.js` client and the `Electron` client?

Reply:
> Was this tested with both, the `NW.js` client and the `Electron` client?

Affirmative. Event loaded a bunch of them . All 5.5 mb pictures (so even more larger data URI)

![image](https://github.com/user-attachments/assets/019c7932-a464-40d7-ab6d-931158eb521d)